File: general_programming_resources/02_code_organization_modularization/examples/messy_autonomous/python/auto.py
import random

# Seeded on purpose so behavior is reproducible before/after refactoring
# (you'll see why that matters in 08_reproducibility).
random.seed(42)


def do_stuff():
    x = read_sensor()
    flag2 = False

    if x > 10:
        y = x * 0.5
        flag2 = True
    else:
        y = x * 2

    print(f"[LOG] sensor={x} output={y}")

    drive_motor(y)

    if flag2:
        print("[LOG] entering hold mode")


def read_sensor():
    return random.random() * 20


def drive_motor(power):
    print(f"Driving motor at power {power}")


if __name__ == "__main__":
    for _ in range(3):
        do_stuff()

File: general_programming_resources/02_code_organization_modularization/examples/messy_autonomous/java/Auto.java
import java.util.Random;

public class Auto {
    // Seeded on purpose so behavior is reproducible before/after refactoring
    // (you'll see why that matters in 08_reproducibility).
    static Random rng = new Random(42);

    public static void doStuff() {
        double x = readSensor();
        double y;
        boolean flag2 = false;

        if (x > 10) {
            y = x * 0.5;
            flag2 = true;
        } else {
            y = x * 2;
        }

        System.out.println("[LOG] sensor=" + x + " output=" + y);

        driveMotor(y);

        if (flag2) {
            System.out.println("[LOG] entering hold mode");
        }
    }

    static double readSensor() {
        return rng.nextDouble() * 20;
    }

    static void driveMotor(double power) {
        System.out.println("Driving motor at power " + power);
    }

    public static void main(String[] args) {
        for (int i = 0; i < 3; i++) {
            doStuff();
        }
    }
}
